sdk/deepbook/src/client.ts

```diff
@@ -298,23 +344,28 @@ export class DeepBookClient {
 			target: `0x2::coin::zero`,
 			arguments: [],
 		});
+
 		const [base_coin_ret, quote_coin_ret] = txb.moveCall({
 			typeArguments: [baseAssetType, quoteAssetType],
 			target: `${PACKAGE_ID}::${MODULE_CLOB}::place_market_order`,
 			arguments: [
 				txb.object(poolId),
-				txb.object(this.#checkAccountCap()),
+				typeof accountCap === 'string' ? txb.object(this.#checkAccountCap(accountCap)) : accountCap,
 				txb.pure.u64(clientOrderId ?? this.#nextClientOrderId()),
 				txb.pure.u64(quantity),
 				txb.pure.bool(orderType === 'bid'),
-				baseCoin ? txb.object(baseCoin) : emptyCoin,
-				quoteCoin ? txb.object(quoteCoin) : emptyCoin,
+				baseCoin ? objArg(txb, baseCoin) : emptyCoin,
+				quoteCoin ? objArg(txb, quoteCoin) : emptyCoin,
 				txb.object(SUI_CLOCK_OBJECT_ID),
 			],
 		});
 		const recipient = this.#checkAddress(recipientAddress);
 		txb.transferObjects([base_coin_ret], recipient);
 		txb.transferObjects([quote_coin_ret], recipient);
+		if (typeof accountCap !== 'string') {
```

Comment:
Don't do this here, just make the consumer do this if they do make the cap in the same transaction